docs/Running-Tachyon-Fault-Tolerant.md

```diff
@@ -0,0 +1,134 @@
+appropriately on each host.
+
+### Externally Visible Address
+
+In the following sections we refer to an "externally visible address". This is simply the address of
+an interface on the machine being configured that can be seen by other nodes in the Tachyon cluster.
+On EC2, you should the `ip-x-x-x-x` address. In particular, don't use `localhost` or `127.0.0.1`, as
+other nodes will then be unable to reach your node.
+
+### Configuring Fault Tolerant Tachyon
+
+In order to enable fault tolerance for Tachyon, additional configuration settings must be set for
+Tachyon masters, workers, and clients. In `conf/tachyon-env.sh`, these java options must be set:
+
+<table class="table">
+<tr><th>Property Name</th><th>Value</th><th>Meaning</th></tr>
+<tr>
+  <td>tachyon.zookeeper.enabled</td>
+  <td>true</td>
+  <td>
+     If true, masters will use ZooKeeper to enable fault tolerance.
+  </td>
+</tr>
+<tr>
+  <td>tachyon.zookeeper.address</td>
+  <td>[zookeeper_hostname]:2181</td>
+  <td>
+    The hostname and port ZooKeeper is running on. Separate multiple addresses with commas.
+  </td>
+</tr>
+</table>
+
+To set these options, you can configure your `TACHYON_JAVA_OPTS` to include:
+
+    -Dtachyon.zookeeper.enabled=true
+    -Dtachyon.zookeeper.address=[zookeeper_hostname]:2181
+
+If you are using a cluster of ZooKeeper nodes, you can specify multiple addresses by separating them
+with commas, like:
+
+     -Dtachyon.zookeeper.address=[zookeeper_hostname1]:2181,[zookeeper_hostname2]:2181,[zookeeper_hostname3]:2181
+
+### Master Configuration
+
+In addition to the above configuration settings, Tachyon masters need additional configuration. The
+following variable must be set appropriately in `conf/tachyon-env.sh`:
+
+    export TACHYON_MASTER_ADDRESS=[externally visible address of this machine]
+
+Also, specify the correct journal folder by setting `tachyon.master.journal.folder` appropriately
+for `TACHYON_JAVA_OPTS`. For example, if you are using HDFS for the journal, you can add:
+
+    -Dtachyon.master.journal.folder=hdfs://[namenodeserver]:[namenodeport]/path/to/tachyon/journal
+
+Once all the Tachyon masters are configured in this way, they can all be started for fault tolerant
+Tachyon. One of the masters will become the leader, and the others will replay the journal and wait
+until the current master dies.
+
+### Worker Configuration
+
+As long as the config parameters above are correctly set, the worker will be able to consult with
+ZooKeeper, and find the current leader master to connect to. Therefore, `TACHYON_MASTER_ADDRESS`
+does not have to be set for the workers.
+
+### Client Configuration
+
+No additional configuration parameters are required for fault tolerant mode. As long as both:
+
+    -Dtachyon.zookeeper.enabled=true
+    -Dtachyon.zookeeper.address=[zookeeper_hostname]:2181
+
+are set appropriately for your client application, the application will be able to consult with
+ZooKeeper for the current leader master.
```

Comment:
it's a bit confusing. In my past experience in using spark, with tachyon-ft, `sc.textFile("tachyon://{one of the master addresses}:19998/file")` will get the data, when the above two parameters are not set in spark job. In the code path, if the master address used is not leader master, then after trying, the request will be transferred to the leader master, I remember it works like this before.